src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -196,37 +196,38 @@ public static IServiceCollection ConfigureCollectionRules(this IServiceCollectio
             return services;
         }
 
-        public static IServiceCollection RegisterCollectionRuleAction<TFactory, TOptions>(this IServiceCollection services, string actionName)
+        public static IServiceCollection RegisterCollectionRuleAction<TFactory, TOptions, TDescriptor>(this IServiceCollection services)
             where TFactory : class, ICollectionRuleActionFactory<TOptions>
             where TOptions : BaseRecordOptions, new()
+            where TDescriptor : class, ICollectionRuleActionDescriptor, new()
         {
             services.AddSingleton<TFactory>();
             services.AddSingleton<CollectionRuleActionFactoryProxy<TFactory, TOptions>>();
-            services.AddSingleton<ICollectionRuleActionDescriptor, CollectionRuleActionDescriptor<TFactory, TOptions>>(sp => new CollectionRuleActionDescriptor<TFactory, TOptions>(actionName));
+            services.AddSingleton<ICollectionRuleActionDescriptor, TDescriptor>(sp => new TDescriptor());
```

Comment:
Do the descriptors actually need to be explicitly constructed? Couldn't service registration just do this for us e.g. `services.AddSingleton<ICollectionRuleActionDescriptor, TDescriptor>()`. Same question for the other descriptor registrations. If not, you can also remove the `new()` constraint on `TDescriptor`.